### ISSUE TITLE
Unicode (UTF-8) problem

### ISSUE THREAD

**Issue by kubik256** [xybu/onedrive-d-old#44]
There is some problem with unicode file names...

`OSError: [Errno 2] Directory or file doesn't exist: '/home/kubik256/OneDrive/Documents/odstoupen\xc3\xad od smlouvy ve 14ti denni lhute.docx'`

Instead of `\xc3\xad` there has to be char `í` ...that makes the fail :(

For now i'll fix that by renaming that file directly inside OneDrive web portal, but for future it will need some fix ;)

Thanks, BR.

**Comment by kubik256** [comment]
Bad luck, after I fix one file there comes another with the same issue.

There is....
`\xc3\xa9` instead of `é`
`\xc3\xad` instead of `í`

...and so on, your script probably don't work with UTF-8 at all :(

**Comment by Vladx71** [comment]
as I said maybe it is an acceptable workaround to skip those files, but instead of this the sync process stops. of course the best solution is to enable syncing of UTF8/16 characters

**Comment by xybu** [comment]
Python 2.x does not handle Unicode file names seamlessly as Python 3 does.
I attempt to encode all local paths with system encode. The commit is 0b86e46b07bbe88eda39650640919f04a201de91

but the API package onedrive-d uses seems to have some issue with file names with Unicode chars. I raised it at https://github.com/mk-fg/python-onedrive/issues/19

Once I get feedback from the writer of python-onedrive I can do some further attempts.

**Comment by kubik256** [comment]
My system (Linux Mint 17 cinnamon 64bit) has default encoding set to UTF-8.
The problem is very strange because some of files with the same chars are synchronized with no error, but only a few of them raise this error.

Thanks for fast reply, BR ;)

**Comment by xybu** [comment]
It is Python 2.x that doesn't treat all strings as Unicode formatted. Refer to http://stackoverflow.com/questions/10180765/open-file-with-a-unicode-filename

When I tested my changes a moment ago, I could not upload my file with Unicode name. Not sure how the API handled the encoding so I raised the issue to the author of the API binding.

**Comment by kubik256** [comment]
Here is some trace-back info for better searching in source code ;)

```
Traceback (most recent call last):
  File "/usr/local/bin/onedrive-cli", line 9, in <module>
    load_entry_point('python-onedrive==14.06.3', 'console_scripts', 'onedrive-cli')()
  File "/usr/local/lib/python2.7/dist-packages/onedrive/cli_tool.py", line 310, in main
    contents = api.get(resolve_path(optz.file), byte_range=optz.byte_range)
  File "/usr/local/lib/python2.7/dist-packages/onedrive/cli_tool.py", line 239, in <lambda>
    resolve_path = ( (lambda s: id_match(s) or resolve_path_wrap(s)) \
  File "/usr/local/lib/python2.7/dist-packages/onedrive/cli_tool.py", line 238, in <lambda>
    resolve_path_wrap = lambda s: api.resolve_path(s and s.replace('\\', '/').strip('/'))
  File "/usr/local/lib/python2.7/dist-packages/onedrive/api_v5.py", line 481, in resolve_path
    raise DoesNotExists(root_id, path[i:])
onedrive.api_v5.DoesNotExists: (u'folder.a6f4d800e6f16741.A6F4D800E6F16741!2035', [u'Odpor proti elektronick\u0102\u0160mu platebn\u0102\xadmu rozkazu.docx'])
Exception in thread Thread-2:
Traceback (most recent call last):
  File "/usr/lib/python2.7/threading.py", line 810, in __bootstrap_inner
    self.run()
  File "/usr/local/lib/python2.7/dist-packages/onedrive_d-0.8-py2.7.egg/onedrive_d/components.py", line 121, in run
    self.consume(task)
  File "/usr/local/lib/python2.7/dist-packages/onedrive_d-0.8-py2.7.egg/onedrive_d/components.py", line 87, in consume
    old_mtime = os.stat(t.p1).st_mtime
OSError: [Errno 2] Adresář nebo soubor neexistuje: '/home/kubik256/OneDrive/Documents/_HONDA/Odpor proti elektronick\xc3\xa9mu platebn\xc3\xadmu rozkazu.docx'
```

Don't worry I always have the same problems with string encodings ;)
Have a nice day and thanks for your previous reply, BR.

**Comment by xybu** [comment]
The Unicode issue is kind of complex. I am putting my updates at https://github.com/mk-fg/python-onedrive/issues/19.

**Comment by Tronic69** [comment]
You said "Python 2.x does not handle Unicode file names seamlessly as Python 3 does". That means filenames will be treted correct, if I have installed Python 3?

I have inastalled Python 2.7.6 and 3.4.0, but when calling just python then 2.7.6 will be used:

$> python -V
Python 2.7.6

$> python3 -V
Python 3.4.0

How could I tell onedrive-d to use python3 instead of python? Maybe that fixes my problems.
